### DISCUSSION TITLE
Remove Forced Login to Access APIs to Unrelated Server

### DISCUSSION THREAD

**Discussion by ryanholden8** [Kong/insomnia#7362] [Ideas]
Currently our team is sitting here unable to use Insomnia to test our APIs that hit against our own local servers. Why? Because we need to login, for no reason.

@NicolaiHerforth said it well https://github.com/Kong/insomnia/issues/7361#issuecomment-2090782117

> Have the same issue. Cannot log in, and therefore cannot access all my requests.
> 
> It seems absolutely insane to me that saved content is behind a login server that could go down or prove problematic...

@sroland-trimble https://github.com/Kong/insomnia/issues/7361#issuecomment-2090871707

> hundreds of people unable to work. This is an insane productivity loss and isn't acceptable

@mikebridge https://github.com/Kong/insomnia/issues/7361#issuecomment-2090894688

> with the migration to the cloud, insomnia has turned itself into a [single point of failure](https://www.techtarget.com/searchdatacenter/definition/Single-point-of-failure-SPOF) for my company.

Some references, all on the issues for a _forced_ login to access local only content:
https://github.com/Kong/insomnia/discussions/7335
https://github.com/Kong/insomnia/issues/7361
https://github.com/Kong/insomnia/issues/7359
https://github.com/Kong/insomnia/issues/7358
https://github.com/Kong/insomnia/discussions/6962 (A response from Insomnia that gives pause for concern on if we should continue using Insomnia)
https://github.com/Kong/insomnia/discussions/6680
https://github.com/Kong/insomnia/discussions/6565
https://github.com/Kong/insomnia/discussions/6590

**Comment by mikebridge** [comment]
Quite apart from the instability of insomnia's cloud infrastructure, we should go back to being able to use this product offline. 

I want to be able to use insomnia in my local dev environment when I can't connect to the internet, like while on a plane, for example.  Right now if I can't connect to the network for some reason, no matter what it is, I'm dead in the water.

**Comment by domigi** [comment]
@subnetmarco Are there any plans in removing the forced account while using insomnia locally? As you mentioned [here](https://github.com/Kong/insomnia/discussions/6626#discussioncomment-7332014) in October, the limit is 30 days which isn't useful for a airgapped environment. 

Also more importantly the systemfailure yesterday (in which you basically bricked any install untill your backend was available again), proved that having this dependency on your cloud is just business unfriendly. How should I be inclined to pay money to a company that makes such technical decisions?

TLDR: Your downtime yesterday proved why forcing users to use a account is not good.

**Reply by subnetmarco** [reply]
Yes, that was not ideal. A unique set a circumstances happened such as we experienced approximately ~2h of downtime. We are putting in place mitigation efforts to prevent this from happening again.

> the limit is 30 days which isn't useful for a airgapped environment

It is 30 days from the last time a session was recognized, so as long as there is Internet connection once every 30 days, that expiration will be pushed back by another 30 days from the last time the session has been refreshed (that is, the last time Insomnia was able to communicate to the servers). We could choose more than 30 days but that would create a security risk for the users.

We are also updating the application in such a way that it can run entirely offline and does not depend on any internet connection for as long as the session is valid (obviously the cloud features will not be available when offline): we pushed many updates in `9.0`, `9.1` and `9.1.1` to achieve this and we are continuing to do so in the following releases. In a quarter from now, this will be a non-issue, as we are literally testing the application while developing in offline mode, we identified a couple more areas of improvement which will be addressed in the coming releases. 

This will have the positive side effect of also removing some performance lag as we are making those cloud calls asynchronous, so the application will be faster both with an internet connection and without one. This has been long overdue.

**Comment by aventus13** [comment]
I have missed the whole changes for the past few months while I'd been using Insomnia on my older machine. Since I switched to a new one, and installed the latest version of Insomnia I was astonished finding out that I have to log in to be able to import collections, as I asked [here](https://github.com/Kong/insomnia/discussions/7335). Honestly it's sad how users are forced to create accounts. In the past me and my team switched from Postman to Insomnia partly for that very reason (another reason was feature bloat).

In my team we immediately decided to replace Insomnia with something else, and something under our control. We're currently trying out bruno- it's very clearly still in beta stage but seems decent so far. And it's open source.

**Reply by subnetmarco** [reply]
> Honestly it's sad how users are forced to create accounts.

I would really like to give Insomnia for free without no hints to monetization, but that is not possible anymore and it is not sustainable in the long term for the project. Insomnia is probably the project that gives the most capabilities out for free already, more than any other alternative like Bruno, which gate keeps upcoming features like GRPC and Websocket behind a paywall.

We are even going as far as making sure that the offline experience is as pleasant as it can be by not requiring to login after the first time and remembering the session, reducing the dependency on the cloud for most operations, and so on.

So, we hope to win you back with our features and with the momentum we are able to now achieve by requiring logins and being able to invest more into the product. Which in turn creates a net positive for both you - as a user of many free capabilities - and us (as a vendor and maintainer of the project who need to sustainably run the business).

As a user of many other projects myself, I find this to be a **fair contract**, I would personally want the projects that I use to be successful. Of course you may think otherwise, and that is also fair. I am not here to convince you, simply giving insights on my worldview.
